python/cudf/cudf/core/column/column.py

```diff
@@ -136,14 +136,11 @@ def to_pandas(self):
         return sr
 
     def clip(self, lo, hi):
-        # If categorical, it returns non categorical column
-        # as there is possibility of change in categories
         if is_categorical_dtype(self):
             input_col = self.astype(self.categories.dtype)
+            return libcudf.replace.clip(input_col, lo, hi).astype(self.dtype)
```

Comment:
If the categories are ordered then we can just clip on the integers based on the encoded value of `lo` / `hi`.

If the categories are unordered then we need to either do this, not cast back to the categorical type, or throw.

Does Pandas always just return a non-categorical type afterwards?

Reply:
Pandas always return non-categorical.

```
If the categories are unordered then we need to either do this, not cast back to the categorical type, or throw.
```
it would be confusing to the user if he can't rely on output type of column.

Reply:
Okay, lets leave the behavior as is for now and we can change to throw later if desired.